scipy/stats/_continuous_distns.py

```diff
@@ -3733,322 +3759,497 @@ class levy_stable_gen(rv_continuous):
+                    return 2. * expr_1 * np.exp(expr_1 * np.tan(theta) / beta
+                                                ) / np.cos(theta) / np.pi
 
                 def g(theta):
                     return np.exp(-np.pi * x / 2. / beta) * V(theta)
 
                 def f(theta):
-                    return g(theta) * np.exp(-g(theta))
+                    g_1 = g(theta)
+                    # as x*exp(-x) -> 0 as x -> inf
+                    return 0 if g_1 == np.inf else g_1 * np.exp(-g_1)
 
                 with np.errstate(all="ignore"):
-                    intg_max = optimize.minimize_scalar(lambda theta: -f(theta), bounds=[-np.pi/2, np.pi/2])
-                    intg = integrate.fixed_quad(f, -np.pi/2, intg_max.x)[0] + integrate.fixed_quad(f, intg_max.x, np.pi/2)[0]
+                    peak = optimize.ridder(
+                        lambda t2: g(t2) - 1, -np.pi / 2, np.pi / 2
+                    )
+
+                    # find better upper bound as
+                    # quadpack adaptive quadrature
+                    # unable to handle long zero tails.
+                    # TODO: better approach?
+                    upper_interval = np.pi / 2 - peak
+                    upper_bound = np.pi / 2
+                    delta_sign = 1
+                    test_point = peak
+                    for i in range(20):
+                        delta = upper_interval / 2.**i
+                        test_point = test_point + delta_sign * delta
+                        if np.isclose(
+                            f(test_point), 0, rtol=1e-013, atol=1e-013
+                        ):
+                            upper_bound = test_point
+                            delta_sign = -1
+                        else:
+                            delta_sign = 1
+
+                    intg = integrate.quad(
+                        f, -np.pi / 2, peak
+                    )[0] + integrate.quad(f, peak, upper_bound)[0]
+
                     return intg / np.abs(beta) / 2.
             else:
-                return 1/(1+x**2)/np.pi
+                return 1 / (1 + x**2) / np.pi
 
     @staticmethod
-    def _cdf_single_value_zolotarev(x, alpha, beta):
-        """Calculate cdf using Zolotarev's methods as detailed in [BS].
+    def _cdf_single_value_piecewise(x, alpha, beta):
+        """Calculate cdf using Nolan's methods as detailed in [NO].
         """
-        zeta = -beta*np.tan(np.pi*alpha/2.)
+        zeta = -beta * np.tan(np.pi * alpha / 2.)
         if alpha != 1:
             x0 = x + zeta  # convert to S_0 parameterization
-            xi = np.arctan(-zeta)/alpha
+            xi = np.arctan(-zeta) / alpha
 
             def V(theta):
-                return np.cos(alpha*xi)**(1/(alpha-1)) * \
-                                (np.cos(theta)/np.sin(alpha*(xi+theta)))**(alpha/(alpha-1)) * \
-                                (np.cos(alpha*xi+(alpha-1)*theta)/np.cos(theta))
+                return np.cos(alpha * xi)**(1 / (alpha - 1)) * (
```

Comment:
This function is a duplicate of `V(theta)` from `_pdf_single_value_piecewise()` in the same module

The crusade against duplication is probably sometimes a bit much, but just bringing this up.

Reply:
I think perhaps on a later refactor this class could go in its own module and common code could be pushed up to the top. 

Regarding amending commits with ENH prefixes, I can't really help there as I'm likely to get it wrong. Sorry.